countComponents.js

```diff
@@ -0,0 +1,75 @@
+/*
+graph of nodes
+
+array[i] = [a, b]
+
+i: n of nodes
+o: => number of connected components
+
+Input: n = 5, edges = [[0,1],[1,2],[3,4]]
+Output: 2
+
+0 - 1 - 2
+
+3 - 4
+
+*/
+
+
+const buildGraph = (edges) => {
+    const graph = {};
+
+    for (const edge of edges) {
+        const [nodeOne, nodeTwo] = edge;
+
+        const isNodeOne = graph.hasOwnProperty(nodeOne);
+        const isNodeTwo = graph.hasOwnProperty(nodeTwo);
+
+        if (!isNodeOne) graph[nodeOne] = [];
+        if (!isNodeTwo) graph[nodeTwo] = [];
+
+        graph[nodeOne].push(nodeTwo);
+        graph[nodeTwo].push(nodeOne);
+    }
+
+    return graph;
+};
+
+const markAsVisited = (node, edges, visited) => {
+    const queue = new Queue();
+    queue.enqueue([node]);
+
+    const graph = buildGraph(edges);
+    while (queue.size() > 0) {
+        // Remove node
+        const [node] = queue.dequeue();
+
+        // Process node
+
+        // Get children
+        const children = graph[node];
```

Comment:
In an undirected graph (which is what we have in this problem), we name these neighbors, not children. Children implies directionality (if `a` is a child of `b`, then it doesn't make sense for `b` to be a child of `a` as well).